src/main/java/com/fasterxml/jackson/databind/deser/std/FactoryBasedEnumDeserializer.java

```diff
@@ -136,10 +136,11 @@ public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
             //   2.11, was just assuming match)
             if (_creatorProps != null) {
                 if (p.isExpectedStartObjectToken()) {
-                    _propCreatorRef.compareAndSet(null,
-                        PropertyBasedCreator.construct(ctxt, _valueInstantiator, _creatorProps,
-                            ctxt.isEnabled(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES)));
-
+                    if (_propCreatorRef.get() == null) {
```

Comment:
Ok, I'll change this afterwards....

Reply:
@JooHyukKim yes, basically I did that; no need to go back and forth. Sorry about confusion.

(the problem with JDK (de)serialization of JsonDeserializers is that we do not want to serialize fields like this, but in case of `AtomicReference` would need to make sure it's not-null ... `volatile` is easier, and performance won't matter a lot here-- although to be completely honest I think lazy construction here is probably not a good idea. Could just create eagerly.

Reply:
Right, just the volatile part.